# Output Equation

## модель 1

# спецификация 
$$
log(output)= \beta_0 + \beta_1 * log(capital) + \beta_2 * log(labour) +u 
$$

In [1]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Labour.csv')
df.head()

capital  labour     output       wage
0   2.606563     184   9.250759  43.080307
1   1.323237      91   3.664310  27.780016
2  22.093692     426  28.781516  44.467748
3  10.737851      72   4.124642  39.734710
4   1.161365      46   2.890150  34.650709

## Cпособ оценивания через матрицы ргерессивного дизайна Y,X

In [3]:
# Зададим матрицы регрессионного дизайна
y = np.log(df['output']) # зависимая переменная
# логаримируем данные
X = pd.DataFrame(data={'l_capital': np.log(df['capital']), 'l_labour': np.log(df['labour'])} )
X = sm.add_constant(X)
#X = sm.add_constant(df[['totwrk', 'male']]) 
X.head(n=7)

const  l_capital  l_labour
0    1.0   0.958033  5.214936
1    1.0   0.280081  4.510860
2    1.0   3.095292  6.054439
3    1.0   2.373775  4.276666
4    1.0   0.149596  3.828641
5    1.0   5.623283  8.941415
6    1.0   0.518174  3.871201

In [4]:
output_eq2 = sm.OLS(y, X).fit()
# Коэфициенты модели с округлением
output_eq2.params.round(3)

const       -1.711
l_capital    0.208
l_labour     0.715
dtype: float64


- увеличении капитала на 1%, выпуск увеличивается на 0.208%
- При увеличении количества рабочих на 1%, выпуска возрастает на 0.715% при прочих равных условиях

## модель 2

$$
log(output)= \beta_0 + \beta_1 * log(capital) + \beta_2 * log(labour) + \beta_3 * log (wage) + u
$$

In [6]:
# Зададим матрицы регрессионного дизайна
y = np.log(df['output']) # зависимая переменная
# логаримируем данные
X = pd.DataFrame(data={'l_capital': np.log(df['capital']), 'l_labour': np.log(df['labour']), 'l_wage': np.log(df['wage'])} )
X = sm.add_constant(X)
#X = sm.add_constant(df[['totwrk', 'male']]) 
X.head(n=7)

const  l_capital  l_labour    l_wage
0    1.0   0.958033  5.214936  3.763066
1    1.0   0.280081  4.510860  3.324317
2    1.0   3.095292  6.054439  3.794764
3    1.0   2.373775  4.276666  3.682225
4    1.0   0.149596  3.828641  3.545318
5    1.0   5.623283  8.941415  3.890591
6    1.0   0.518174  3.871201  3.479950

In [7]:
output_eq2 = sm.OLS(y, X).fit()
# Коэфициенты модели с округление
output_eq2.params.round(3)

const       -5.007
l_capital    0.149
l_labour     0.720
l_wage       0.921
dtype: float64

- При увеличении количества рабочих на 1%, выпуск возрастает на 0.72%
- При увеличении капитала на 1%, выпуск увеличивается на 0.149%, при прочих равных